mlir/python/mlir/dialects/_ods_common.py

```diff
@@ -78,12 +78,12 @@ def equally_sized_accessor(
 def get_default_loc_context(location=None):
     """
     Returns a context in which the defaulted location is created. If the location
-    is None, takes the current location from the stack, raises ValueError if there
-    is no location on the stack.
+    is None, takes the current location from the stack.
     """
     if location is None:
-        # Location.current raises ValueError if there is no current location.
-        return _cext.ir.Location.current.context
+        if _cext.ir.Location.current:
+            return _cext.ir.Location.current.context
+        return None
```

Comment:
I realized since we're now supporting loc inference we shouldn't require a location to be in the context; specifically `Location.current` shouldn't throw but just return `None`. See tests in `context_manager.py` below.

Reply:
@jpienaar @rolfmorel i'm asking for a re-review because I made this change which wasn't in the original PR and slightly changes semantics; see also the change to `Location.current` here https://github.com/llvm/llvm-project/pull/151246/files#r2268615944

Reply:
SGTM